cli_test.test_verbose is flaky depending on fork race

```
21:49:40 =================================== FAILURES ===================================
21:49:40 _______________________________ test_verbose[-v] _______________________________
21:49:40 
21:49:40 flag = '-v'
21:49:40 
21:49:40     @pytest.mark.parametrize('flag', ['-v', '--verbose'])
21:49:40     def test_verbose(flag):
21:49:40         """dumb-init should print debug output when asked to."""
21:49:40         proc = Popen(('dumb-init', flag, 'echo', 'oh,', 'hi'), stdout=PIPE, stderr=PIPE)
21:49:40         stdout, stderr = proc.communicate()
21:49:40         assert proc.returncode == 0
21:49:40         assert stdout == b'oh, hi\n'
21:49:40 >       assert re.match(
21:49:40             (
21:49:40                 b'^\[dumb-init\] Child spawned with PID [0-9]+\.\n'
21:49:40                 b'\[dumb-init\] setsid complete\.\n'
21:49:40                 b'\[dumb-init\] Received signal 17\.\n'
21:49:40                 b'\[dumb-init\] A child with PID [0-9]+ exited with exit status 0.\n'
21:49:40                 b'\[dumb-init\] Forwarded signal 15 to children\.\n'
21:49:40                 b'\[dumb-init\] Child exited with status 0\. Goodbye\.\n$'
21:49:40             ),
21:49:40             stderr,
21:49:40         ), stderr
21:49:40 E       AssertionError: [dumb-init] setsid complete.
21:49:40 E         [dumb-init] Child spawned with PID 12915.
21:49:40 E         [dumb-init] Received signal 17.
21:49:40 E         [dumb-init] A child with PID 12915 exited with exit status 0.
21:49:40 E         [dumb-init] Forwarded signal 15 to children.
21:49:40 E         [dumb-init] Child exited with status 0. Goodbye.
21:49:40 E         
21:49:40 E       assert None
21:49:40 E        +  where None = <function match at 0x2af9ace03c80>('^\\[dumb-init\\] Child spawned with PID [0-9]+\\.\n\\[dumb-init\\] setsid complete\\.\n\\[dumb-init\\] Received signa...us 0.\n\\[dumb-init\\] Forwarded signal 15 to children\\.\n\\[dumb-init\\] Child exited with status 0\\. Goodbye\\.\n$', '[dumb-init] setsid complete.\n[dumb-init] Child spawned with PID 12915.\n[dumb-init] Received signal 17.\n[dumb-init]... with exit status 0.\n[dumb-init] Forwarded signal 15 to children.\n[dumb-init] Child exited with status 0. Goodbye.\n')
21:49:40 E        +    where <function match at 0x2af9ace03c80> = re.match
21:49:40 
21:49:40 tests/cli_test.py:85: AssertionError
```

--- a/tests/child_processes_test.py
+++ b/tests/child_processes_test.py
@@ -91,8 +91,7 @@ def spawn_process_which_dies_with_children():
     # read a line from print_signals, figure out its pid
     line = proc.stdout.readline()
     match = re.match(b'ready \(pid: ([0-9]+)\)\n', line)
-    assert match, 'print_signals should print "ready" and its pid, not ' + \
-        str(line)
+    assert match, line
     child_pid = int(match.group(1))
 
     # at this point, the shell and dumb-init have both exited, but
@@ -139,9 +138,9 @@ def test_processes_dont_receive_term_on_exit_if_no_setsid():
 def test_fails_nonzero_with_bad_exec(args):
     """If dumb-init can't exec as requested, it should exit nonzero."""
     proc = Popen(('dumb-init',) + args, stderr=PIPE)
-    proc.wait()
+    _, stderr = proc.communicate()
     assert proc.returncode != 0
     assert (
         b'[dumb-init] /doesnotexist: No such file or directory\n'
-        in proc.stderr
+        in stderr
     )

--- a/tests/cli_test.py
+++ b/tests/cli_test.py
@@ -82,16 +82,20 @@ def test_verbose(flag):
     stdout, stderr = proc.communicate()
     assert proc.returncode == 0
     assert stdout == b'oh, hi\n'
-    assert re.match(
+
+    # child/parent race to print output after the fork(), can't guarantee exact order
+    assert re.search(b'(^|\n)\[dumb-init\] setsid complete\.\n', stderr), stderr  # child
+    assert re.search(  # parent
         (
-            b'^\[dumb-init\] Child spawned with PID [0-9]+\.\n'
-            b'\[dumb-init\] setsid complete\.\n'
+            b'(^|\n)\[dumb-init\] Child spawned with PID [0-9]+\.\n'
+            b'.*'  # child might print here
             b'\[dumb-init\] Received signal 17\.\n'
             b'\[dumb-init\] A child with PID [0-9]+ exited with exit status 0.\n'
             b'\[dumb-init\] Forwarded signal 15 to children\.\n'
             b'\[dumb-init\] Child exited with status 0\. Goodbye\.\n$'
         ),
         stderr,
+        re.DOTALL,
     ), stderr
 
 

